adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m

```diff
@@ -127,25 +127,49 @@ - (void)addDelegate:(nonnull id<GADMAdapterVungleDelegate>)delegate {
     }
   } else if ([delegate respondsToSelector:@selector(isBannerAd)] && [delegate isBannerAd]) {
     @synchronized(_bannerDelegates) {
-      _bannerPlacementID = [delegate.desiredPlacement copy];
-      if (delegate && ![_bannerDelegates objectForKey:delegate.desiredPlacement]) {
-        GADMAdapterVungleMapTableSetObjectForKey(_bannerDelegates, delegate.desiredPlacement,
-                                                 delegate);
+      if (delegate && ![_bannerDelegates objectForKey:delegate.bannerRequest]) {
+        NSEnumerator *enumerator = _bannerDelegates.keyEnumerator;
+        GADMAdapterVungleBannerRequest *bannerRequest = nil;
+        while (bannerRequest = [enumerator nextObject]) {
+          if ([bannerRequest.placementID isEqualToString:delegate.bannerRequest.placementID]) {
+            return NO;
+          }
+        }
+
+        NSMutableOrderedSet *set = [[NSMutableOrderedSet alloc] initWithObject:delegate];
+        GADMAdapterVungleMapTableSetObjectForKey(_bannerDelegates, delegate.bannerRequest,
+                                                 set);
+        _bannerPlacementID = [delegate.desiredPlacement copy];
+      } else if (delegate && [_bannerDelegates objectForKey:delegate.bannerRequest]) {
```

Comment:
@ashinagawa Hi Aki, `[_bannerDelegates objectForKey:delegate.bannerRequest]` is checking `bannerRequest` object, which means it will check placementID and UUID. 
In `GADMAdapterVungleBannerRequest` class, I ovewrite the `- (BOOL)isEqual:` and `- (NSUInteger)hash` methods, which are used for `NSMapTable` search operation and determine if two `GADMAdapterVungleBannerRequest` objects are equal. If two `GADMAdapterVungleBannerRequest` objects have same placementID and UUID, they are equal even though they are different objects( memory addresses are different). Thanks 🙂

Reply:
@yueluliu-vungle Ah, I got it. I missed to check overwritten `- (BOOL)isEqual:` method.
Thanks for clarifying it.